packages/rum-core/src/domain/assembly.ts

```diff
@@ -113,30 +115,27 @@ export function startRumAssembly(
             id: configuration.applicationId,
           },
           date: timeStampNow(),
-          service: configuration.service,
+          service: isExperimentalFeatureEnabled('sub-apps')
+            ? viewContext.service || configuration.service
+            : configuration.service,
+          version: isExperimentalFeatureEnabled('sub-apps') ? viewContext.version || configuration.version : undefined,
           source: 'browser',
           session: {
             id: session.id,
             type: syntheticsContext ? SessionType.SYNTHETICS : ciTestContext ? SessionType.CI_TEST : SessionType.USER,
           },
+          view: {
+            id: viewContext.view.id,
+            name: viewContext.view.name,
+            url: urlContext.view.url,
+            referrer: urlContext.view.referrer,
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏maybe remove the `view` extra-level from those contexts